slopmop/sm.py

```
@@ -1098,6 +1013,20 @@ def main_error(message: str) -> None:
     return parser
 
 
+def _disable_tty_stdout() -> None:
+    """Wrap stdout so it reports non-TTY, and disable color output."""
+    os.environ["NO_COLOR"] = "1"
+
+    class _NT:
+        def isatty(self) -> bool:
+            return False
+
+        def __getattr__(self, n: str, _o: Any = sys.stdout) -> Any:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_o`

(ANN401)